include/uv-darwin.h

```diff
@@ -52,6 +57,7 @@
   void* cf_member[2];                                                         \
   int cf_error;                                                               \
   uv_mutex_t cf_mutex;                                                        \
+  enum uv_fs_event_type_flag type_flag;                                       \
```

Comment:
Well, you could just put everything but `basename` to the `->realpath` property of the handle. It will automatically do the same thing.